questionnaire/dataset/dataset-service.js

```diff
@@ -21,42 +72,84 @@ function createDatasetService({
                 Object.keys(sectionAnswers).forEach(questionId => {
                     const answer = sectionAnswers[questionId];
+                    const existingAnswer = dataset.get(questionId);
+
+                    if (existingAnswer !== undefined) {
+                        const mergedAnswers = mergeArrays(existingAnswer, answer);
 
-                    if (dataset.has(questionId)) {
-                        const existingAnswer = dataset.get(questionId);
-
-                        // Answers for a specific question id can be collected over multiple sections.
-                        // If previous answers have already be processed for an id, use an appropriate
-                        // merge strategy for the given data type e.g. push to an array, concatenate strings with a delimiter
-                        // ONLY SUPPORTS ARRAYS AT THE MOMENT
-                        if (Array.isArray(existingAnswer)) {
-                            // Ensure answers are of the same type
-                            if (Array.isArray(answer)) {
-                                existingAnswer.push(...answer);
-                            } else {
-                                throw new VError(
-                                    `Question id "${questionId}" found more than once with different answer types. Unable to combine type "array" with "${typeof answer}"`
-                                );
-                            }
-                        } else {
-                            throw new VError(
-                                `Question id "${questionId}" found more than once with unsupported type "${typeof existingAnswer}". Only arrays can be used to combine answers for a single id`
-                            );
-                        }
+                        dataset.set(questionId, mergedAnswers);
                     } else {
                         dataset.set(questionId, answer);
                     }
                 });
             }
         });
 
-        return [
-            {
-                type: 'dataset',
-                id: 0,
-                attributes: Object.fromEntries(dataset)
+        return Object.fromEntries(dataset);
+    }
+
+    function getHierachicalDataView(questionnaire) {
+        const {progress, sections, answers} = questionnaire;
+        const dataset = new Map();
+
+        progress.forEach(sectionId => {
+            const questionAnswers = answers[sectionId];
+
+            if (questionAnswers !== undefined) {
+                const sectionDefinition = sections[sectionId];
+                const section = createSection({sectionDefinition});
+                const attributes = section.getAttributesByData(questionAnswers);
+
+                attributes.forEach(attribute => {
+                    const existingAttribute = dataset.get(attribute.id);
+
+                    if (existingAttribute !== undefined) {
+                        mergeAttributeValues(existingAttribute, attribute);
```

Comment:
is this the functionality specifically for the merging physical injuries answers from multiple pages?

Reply:
Yeah, just any attributes that might be captured across multiple pages. Only physical injuries at the moment